onadata/apps/viewer/models/data_dictionary.py

```diff
@@ -78,15 +78,10 @@ def set_uuid_in_xml(self, file_name=None, id_string=None):
         doc = clean_and_parse_xml(self.xml)
         model_nodes = doc.getElementsByTagName("model")
         if len(model_nodes) != 1:
-            node_counts = 0
             for node in model_nodes:
-                if hasattr(node, 'firstChild') and hasattr(node.firstChild, 'nodeName'):
-                    if node.firstChild.nodeName == 'instance':
-                        if node_counts != 0:
-                            raise Exception("xml contains multiple model nodes")
-                        else:
-                            model_node = node
-                            node_counts += 1
+                if hasattr(node, 'parentNode') and node.parentNode == 'h:head':
```

Comment:
I think we need to use the `tag name` here (i.e.: `node.parentNode.tagName  == 'h:head'`). 

`parentNode` is a miniDom object and does not equal 'h:head`. Therefore, model_node is not set and an error is raised on line 87 

> UnboundLocalError: local variable 'model_node' referenced before assignment